### PR TITLE
Adding protobuf support

### PR DESCRIPTION
**Pull request by suhrm**

### CODE REVIEW

File: protobuf/abacus_metric.proto [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        uint64 uint64_max = 9;
+        int64 int64_max = 10;
+        double float64_max = 11;
+    }
+}
+
+message Metric
+{
+    Info info = 1;
```
Comment (techbech) [inline protobuf/abacus_metric.proto:42]:
```suggestion
    optional Info info = 1;
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -57,6 +61,7 @@ else()
 endif()
 
 target_link_libraries(abacus PRIVATE steinwurf::endian)
+target_link_libraries(abacus PUBLIC protobuf::libprotobuf)
```
Comment (techbech) [inline CMakeLists.txt:64]:
```suggestion
target_link_libraries(abacus PRIVATE protobuf::libprotobuf)
```
Don't let protobuf be exposed

**Review by techbech — CHANGES_REQUESTED** [review]
Nice :+1:

**Comment by mortenvp** [comment]
looks good to me :+1:

**Pull request merged**